pyanaconda/ui/lib/disks.py

```diff
@@ -92,3 +92,33 @@ def applyDiskSelection(storage, data, use_names):
 
     data.ignoredisk.onlyuse = onlyuse
     data.clearpart.drives = use_names[:]
+
+def getRelatedDisks(storage, disk):
+    """ Return a list of disks related to disk by way of container devices. """
+    return set(d for dep in storage.deviceDeps(disk) for d in dep.disks)
+
+def checkDiskSelection(storage, all_disks, selected_disks):
+    """ Return a list of errors related to a proposed disk selection.
+
+        :param :class:`blivet.Blivet` storage: storage data
+        :param all_disks: all known disks
+        :type all_disks: list of :class:`blivet.devices.StorageDevice`
+        :param selected_disks: proposed selected disks
+        :type selected_disks: list of :class:`blivet.devices.StorageDevice`
+        :returns: a list of error messages
+        :rtype: list of str
+    """
+    errors = []
+    for selected in all_disks:
+        if selected.name not in selected_disks:
+            continue
+
+        related = getRelatedDisks(storage, selected)
+        missing = [r.name for r in related if r.name not in selected_disks]
+        if missing:
```

Comment:
``if any(missing)`` would feel more natural here to me, but that's just a matter of opinion.